pkg/controller/enterprisesearch/config.go

```diff
@@ -251,16 +253,32 @@ func associationConfig(c k8s.Client, ent entv1beta1.EnterpriseSearch) (*settings
 		return settings.NewCanonicalConfig(), nil
 	}
 
+	// origin of authenticated ent users setting changed starting 8.x
+	cfg := settings.MustCanonicalConfig(map[string]string{
+		"ent_search.auth.source": "elasticsearch-native",
+	})
+	ver, err := version.Parse(ent.Spec.Version)
+	if err != nil {
+		return nil, err
+	}
+	if ver.IsSameOrAfter(version.From(8, 0, 0)) {
+		cfg = settings.MustCanonicalConfig(map[string]string{
+			"ent_search.auth.native1.source": "elasticsearch-native",
+			"ent_search.auth.native1.order": "-100",
```

Comment:
I'm curious why `-100` and not eg. `0`?

Reply:
The order of the realm is set to -100 to favor the fact that it is
used first but there is no complete guarantee about that. The user can
still configure a realm with an order less than -100 but it is very
unlikely given this is not suggested in the documentation (order starts to 0).

(as in https://github.com/elastic/cloud-on-k8s/pull/1698)